pkg/apis/provisioning/v0alpha1/types.go

```diff
@@ -28,30 +28,25 @@ type LocalRepositoryConfig struct {
 type Workflow string
 
 const (
-	// BranchWorkflow creates a branch for changes
+	// WriteWorkflow allows a user to write directly to the repository
+	WriteWorkflow Workflow = "write"
+	// BranchWorkflow allows creating a branch for changes (where we can then make a PR)
 	BranchWorkflow Workflow = "branch"
-	// PushWorkflow pushes changes directly the configured branch
-	PushWorkflow Workflow = "push"
 )
 
 type GitHubRepositoryConfig struct {
 	// The repository URL (e.g. `https://github.com/example/test`).
 	URL string `json:"url,omitempty"`
 
 	// The branch to use in the repository.
-	// By default, this is the main branch.
-	Branch string `json:"branch,omitempty"`
+	Branch string `json:"branch"`
```

Comment:
Why this change? didn't we want to default to `main`?

Reply:
If that's not the case, let's remove the default in the mutator. It doesn't make sense anymore. 

(the whole point is that the OpenAPI spec says this one is now required)

Reply:
ugg -- this is one of the places I wish there was a different spec for what you write vs what you read 😢 

We should *always* save a branch, and it should not be optional in the typescript -- so whatever we need for that 🤷🏻